libglusterfs/src/inode.c

```diff
@@ -917,6 +938,15 @@ __is_root_gfid(uuid_t gfid)
     return _gf_false;
 }
 
+static gf_boolean_t
+__is_root_with_ref(inode_t *inode)
+{
+    if (__is_root_gfid(inode->gfid) && inode->ref)
```

Comment:
As opposed to the gfid, which is not going to change, can't the ref change, if you are testing it not under the lock?

Reply:
Except after creation, root inode should always have >= 1 references. After creation I think we can avoid calling this function, so the check is not needed.